webhook/main.py

```diff
@@ -1,39 +1,46 @@
-from enum import Enum
-from fastapi import FastAPI, Request, Header, HTTPException, Body
-from pydantic import BaseModel
-from typing import List, Optional, Dict, Union
+from fastapi import FastAPI, Header, HTTPException, status
+from google.cloud import ndb
+from typing import Optional, Dict
+
+from models import VTAPI, EmailWrapper, UserEmail
 
 app = FastAPI()
+client = ndb.Client()
 
 
 @app.get("/")
 async def root():
     return {"data": "Hello World"}
 
 
-class VTType(str, Enum):
-    file = 'file'
-    url = 'url'
-    domain = 'domain'
-    ip_address = 'ip_address'
-
-
-class VTData(BaseModel):
-    attributes: Dict
-    id: str
-    links: Dict
-    type: VTType
-
-
-class VTAPI(BaseModel):
-    data: List[VTData]
-    links: Dict
-    meta: Dict
-
-
 @app.post("/query-results/")
 async def query_results(data: VTAPI, x_appengine_inbound_appid: Optional[str] = Header(None)):
     if x_appengine_inbound_appid != 'virustotal-step-2020':
-        raise HTTPException(status_code=403, detail="Access forbidden")
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN, detail="Access forbidden")
 
     return data
+
+
+@ndb.transactional
+def update_email(content: EmailWrapper):
+    with client.context():
+        q = UserEmail.query(api_key=content.api_key)
+        q_result = list(q.fetch())
+        if len(q_result) > 1:
+            raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                                detail="There exist multiple rows with same api key")
```

Comment:
Why check this on every request? This can be expensive and slow.

Reply:
Well there should never occur situation when q_result has more then 1 row and I need to check if given API key exists in datastore so how can I do it better?

Reply:
Yes, you can do the following
```python
q = UserEmail.get_by_id(apikey)
```

Assuming you're using the API key as ID in datastore. Using email would also be valid.

Reply:
Done